client/src/components/Overview.jsx

```diff
@@ -0,0 +1,215 @@
+      })
+      var rating = sum/(res.data.results.length)
+      this.setState({rating: rating})
+    })
+
+    axios.get('https://app-hrsei-api.herokuapp.com/api/fec2/hr-rfp/products/40344', {
+      headers: {
+        authorization: token
+      }
+    })
+    .then(res => {
+      this.setState({productName: res.data.name})
+      this.setState({productCategory: res.data.category})
+      this.setState({productCategory: res.data.category})
+      this.setState({productPrice: res.data.default_price})
+
+    })
+
+
+
+    axios.get('https://app-hrsei-api.herokuapp.com/api/fec2/hr-rfp/products/40344/styles', {
+      headers: {
+        authorization: token
+      }
+    })
+    .then(res => {
+
+
+      // maintain styles API results in state, so that I can use it later in lifecycle
+      this.setState({results: res.data.results})
+
+
+      // initialize available sizes of default style
+      var style = res.data.results[0]
+      var styleSkus = style.skus;
+      var sizesArr = [];
+      for (var sku in styleSkus) {
+        if (!sizesArr.includes(styleSkus[sku].size)) {
+          sizesArr.push(styleSkus[sku].size)
+        }
+      }
+      console.log(sizesArr)
+      this.setState({sizes: sizesArr})
+
+
+      // initialize quantity of default style and selected size
+      var style = res.data.results[0]
+      var styleSkus = style.skus
+      console.log('styleSkus', styleSkus)
+      var quantity = 0;
+      for (var sku in styleSkus) {
+        console.log('state',this.state.selectedSize)
+        console.log(styleSkus[sku].size)
+        if (styleSkus[sku].size === this.state.selectedSize) {
+          quantity += styleSkus[sku].quantity
+        }
+      }
+      this.setState({quantity:quantity})
+
+
+      // render thuumbnails and available sizes of default style
+
+      var urls = [];
+      console.log(res.data.results[0])
+
+      res.data.results.forEach( (style, i) => {
+        urls.push(   {url: style.photos[0]['thumbnail_url'], index: i})
+      })
+      this.setState({thumbnails: urls})
+    })
+
+
+  }
+
+
+  handleStyleClick(e) {
+
+    // update available sizes upon clicking new style
+    this.setState({selectedStyle : event.target.getAttribute("id")}, () => {
+    var style = this.state.results[Number(this.state.selectedStyle)];
+    var styleSkus = style.skus;
+    var sizesArr = [];
+    for (var sku in styleSkus) {
+      if (!sizesArr.includes(styleSkus[sku].size)) {
+        sizesArr.push(styleSkus[sku].size)
+      }
+    }
+    console.log(sizesArr)
+    this.setState({sizes: sizesArr})
+
+
+    // toggle "outOfStock" boolean if quantity of currently selected style === 0
+
+    var quantity = 0;
+      for (var sku in styleSkus) {
+        quantity += styleSkus[sku].quantity
+      }
+      if (quantity === 0) {
+        this.setState({outOfStock : true})
+      } else {
+        this.setState({outOfStock : null})
+      }
+    })
+  }
+
+
+  handleSizeSelect(e) {
+    console.log(e.target.value)
```

Comment:
you have a lot of console.log statement peppered throughout this file. if you still need them comment them out , otherwise delete them.